app/src/theme/utils.ts

```diff
@@ -0,0 +1,22 @@
+import { getLogger } from '../util/logger'
+
+import theme from './index'
+
+const logger = getLogger('Theme::Utils')
+
+export const getOutcomeColor = (index: number): Record<string, any> => {
+  const localTheme: Record<string, any> = theme
+  const defaultReturn = { darker: '#000', medium: '#999' }
+
+  if (!localTheme || !localTheme['outcomes'] || !localTheme['outcomes']['colors']) {
+    logger.error('Error when using theme or theme.outcomes')
+    return defaultReturn
+  }
+
+  const colors = localTheme.outcomes.colors
+  const maxIndex = colors.length - 1
+  const groupIndex = Math.trunc(index / colors.length)
+  const outcomeIndex: number = index - groupIndex - maxIndex * groupIndex
+
+  return colors[outcomeIndex]
```

Comment:
Can't you just do `colors[index % colors.length]`?

Reply:
Don't know, that's the best I could come up with.

Reply:
There's also a problem with `colors.length` not being the same as the array's top index (not sure if your comment addresses that or not).